app/src/controllers/object.js

```diff
@@ -43,6 +43,57 @@ const controller = {
     }
   },
 
+  /**
+   * @function addMetadata
+   * Creates a new version of the object via copy with the new metadata added
+   * @param {object} req Express request object
+   * @param {object} res Express response object
+   * @param {function} next The next callback function
+   * @returns {function} Express middleware function
+  */
+  async addMetadata(req, res, next) {
+    try {
+      const objId = addDashesToUuid(req.params.objId);
+      const objPath = getPath(objId)
+
+      const latest = await storageService.headObject({ filePath: objPath });
+      if (latest.ContentLength > 5368709120) {
+        throw new Error('Cannot copy an object larger than 5GB');
+      }
+      else {
+        if (Object.keys(req.query).length <= 0 ||
+          Object.keys(req.query).length === 1 && req.query.hasOwnProperty('versionId')) {
+          // 422 when no keys given
+          res.status(422).end();
+        }
+        else {
+          const metadataToAppend = req.query;
+
+          if (metadataToAppend.hasOwnProperty('versionId')) delete metadataToAppend['versionId'];
+
+          const data = {
+            copySource: objPath,
+            filePath: objPath,
+            metadata: {
+              ...latest.Metadata,  // Take existing metadata first
+              ...metadataToAppend, // Append new metadata
+              name: latest.Metadata.name,  // Always enforce name and id key behavior
+              id: latest.Metadata.id
+            },
+            metadataDirective: 'REPLACE',
+            versionId: req.query.versionId ? req.query.versionId.toString() : undefined
```

Comment:
If you use the object destructuring comment from earlier, this could be changed to something simpler:
```js
versionId: versionId ? versionId.toString() : undefined
```